### ISSUE TITLE
Training LaneGCN in nuplan_mini is too slow

### ISSUE THREAD

**Issue by CrisCloseTheDoor** [motional/nuplan-devkit#110]
Hi developer:

I found training the lanegcn model(vector) on the mini set(only ~8G) is too slow: each epoch takes **near 6 hours** \. My device: 4* RTX3090, each GPU has 24G memory but the **max** batch size I can use is only 3.

Is there any settings to improve the training efficiency apparently?

**Comment by patk-motional** [comment]
Hi @CrisCloseTheDoor,

I'm sorry for the late reply. We've been gearing up for the competition launch this past few weeks.

This is not what we have experienced. Can you share the command you use to train your model?

Are you using limit_total_scenarios=0.9999 , if so, you will have all the samples in the mini split and might take longer, so perhaps you can try limit_total_scenarios=0.1 or limit_total_scenarios=0.05 during caching to sample less densely while (presumably) still retain decent amount of scenario variation. Secondly, it seems  you previously set precision to 32 in this other [issue](https://github.com/motional/nuplan-devkit/issues/91#issuecomment-1212713059) which might have made training slower as well. An alternative is to use precision=16and start with a smaller learning rate (in my experience 5e-5 works). This might allow the training to stabilise first. For this, we are working on including a warm up phase to the lr scheduler so that training is more stable.
The issue with max batch size being 3 has always been there, whereby >3 might cause OOM. One way to improve this might be to reduce the lanegcn model complexity by changing the architecture by reducing for example the number of attention layers in nuplan_devkit/nuplan/planning/script/config/common/model/vector_model.yaml

**Comment by CrisCloseTheDoor** [comment]
Thanks for your reply, I did use all of the samples to avoid overfitting, thus FP16 with 'nan' problems showed and training process spent a lot of time. I will try limit_total_scenarios=0.1 if it doesn't reduce the performance of the model.

**Comment by patk-motional** [comment]
I forgot to mention, do try to cache your features as well
